### ISSUE TITLE
Training network on GPU. Error: "The JSON-RPC connection with the remote party was lost before the request could complete."

### ISSUE THREAD

**Issue by Exterminated** [dotnet/machinelearning-modelbuilder#1712]
**System Information (please complete the following information):**
 - Model Builder or CLI Version: 16.6.3.2136603
 - Model Builder GPU Support Version: 16.6.3.2136603
 - Visual Studio Version (if applicable): Professional 2019 version 16.11.1
 - .Net 5.0

**Describe the bug**
- On which step of the process did you run into an issue: Train step in Image classification scenario using Local GPU and default flower photos from tutorioal
- Clear description of the problem:

I am trying to take the first steps in a home project, I decided to try to train the network using my local video card, but I ran into the problem "The JSON-RPC connection .."
After searching Google for information, Microsoft recognized this error and they fixed it in 16.5.4, but my environment is newer.
I tried to restore the installation via Visual Studio Installer, but to no avail.
Has anyone encountered such a problem? Can someone suggest how to deal with it.

**To Reproduce**
1. Go to 'Add..'->Machine Learning
2. Click on 'Image classification'
3. Select on Environment tab Local (GPU)
4. Click on "Check combability"
5. Click "Next step" and select dataset (i'm using default dataset with flowers classification) and click "Next step"
6. On Train step click "Train" and wait few seconds
7. See error

**Expected behavior**
Training proceeded without error and i can move to "Evaluate" step

**Screenshots**
![image](https://user-images.githubusercontent.com/4963117/130353273-09437c28-2f1e-44f8-866e-cf4e05ad51bf.png)
![image](https://user-images.githubusercontent.com/4963117/130353289-541a0475-f98f-4d01-9871-ca46ead05663.png)
![image](https://user-images.githubusercontent.com/4963117/130353295-568f6258-b34c-4e64-89b7-a54ade38777d.png)

**Additional context**
[06428ff7-b6fc-4b0a-88b4-4156f15cf916.txt](https://github.com/dotnet/machinelearning-modelbuilder/files/7027246/06428ff7-b6fc-4b0a-88b4-4156f15cf916.txt)
[c652506b-4dd5-4f80-8abe-efbc08b38f84.txt](https://github.com/dotnet/machinelearning-modelbuilder/files/7027247/c652506b-4dd5-4f80-8abe-efbc08b38f84.txt)
[9754cd7b-5563-44ff-abfd-8f684fbf7c68.txt](https://github.com/dotnet/machinelearning-modelbuilder/files/7027248/9754cd7b-5563-44ff-abfd-8f684fbf7c68.txt)

**Comment by beccamc** [comment]
@LittleLittleCloud We have another `The JSON-RPC connection with the remote party was lost before the request could complete` error. This one is on Dev16. Do you have any ideas?

**Comment by LittleLittleCloud** [comment]
The error `The JSON-RPC connection with the remote party was lost before the request could complete error` simply indicates that the training process was terminated unexpectedly. From the log provided, it seems that the training process was terminated after `RawByteImageLoading` and before `image classification`, which is probably where training is broken.

The reason why the training process being terminated can be various. It could be OOM in system, GPU driver issue or other things. Based on the log provided, there's little way for us to find out which is the root cause.

It would be of great help if you can rule out a few possible causes from framework side by running the project below. It's basically a call into MLNet image classification API, which is the same one used by Model Builder in image classification scenario.

**Comment by Exterminated** [comment]
Thank you for responding to my problem! I started the project from the last comment. It looks fine
Results in file
[imageClassificationSample-24-08-21.txt](https://github.com/dotnet/machinelearning-modelbuilder/files/7037080/imageClassificationSample-24-08-21.txt)

**Comment by LittleLittleCloud** [comment]
@Exterminated 

Thanks for the response. So it seems that mlnet framework itself doesn't have problem. And let's try if we can rule out the automl service as well by running the two experiments below:

### First experiment

Try CPU training in ModelBuilder by selecting CPU in Environment step

### Second experiment

Try image classification in mlnet cli.
- install mlnet cli in private channel by running the following command
`dotnet tool install -g mlnet --add-source https://mlnetcli.blob.core.windows.net/mlnetcli/index.json`
- launch image classification
`mlnet image-classification --dataset /path/to/your/image/folder`

If both experiments can be run without problem, we can assume that the error might be in servicehub side

**Comment by Exterminated** [comment]
Yep, both experiments was successful. What i should check/reinstall with servicehub side?
[mlnet-image-test-25-08-2021.txt](https://github.com/dotnet/machinelearning-modelbuilder/files/7045809/mlnet-image-test-25-08-2021.txt)

**Comment by Exterminated** [comment]
@LittleLittleCloud 
Dose i need start training via cmd?
> mlnet image-classification --dataset /path/to/your/image/folder

I replaced dll's and then start training via cmd. I don't see any mentions of my gpu in logs
[mlnet-29-08-2021.txt](https://github.com/dotnet/machinelearning-modelbuilder/files/7072545/mlnet-29-08-2021.txt)

**Comment by LittleLittleCloud** [comment]
@Exterminated 
What you do looks correct. But from the log it seems that the mlnet cli is still using cpu to train the model.

**Comment by LittleLittleCloud** [comment]
Hi @Exterminated 

What's the largest size of your images, the error could be caused by OOM on GPU and is most likely because of some very large pictures.

**Comment by beccamc** [comment]
@Exterminated Is this issue resolved?

**Comment by Exterminated** [comment]
Hi @beccamc! Unfortunately no =(
I was trying retest this a few month ago and get same errors with small pictures.

**Comment by beccamc** [comment]
Have you updated to latest version 16.13.1?

**Comment by Exterminated** [comment]
@beccamc Yes
![image](https://user-images.githubusercontent.com/4963117/160788516-a6cc1d69-72b6-41b3-9163-e599b99f525b.png)
![image](https://user-images.githubusercontent.com/4963117/160788577-0a01a418-93a2-4f04-a9cf-e217b965afde.png)

**Comment by beccamc** [comment]
@LittleLittleCloud Are you able to help here again? It's still an issue.

**Comment by LittleLittleCloud** [comment]
Yeah sure,, we no longer use image classification api from AutoML.Net since v16.13.1 so it might provides more logs and more information to trace. @Exterminated would you be able to share with us model builder log again.

Regards!

**Comment by beccamc** [comment]
I know this is an old issue. We've had a few releases. @Exterminated Any chance you can try on [16.14.0 ](https://marketplace.visualstudio.com/items?itemName=MLNET.ModelBuilder2022) and share logs?
